src/BuildScriptGeneratorCli/ServiceProviderBuilder.cs

```diff
@@ -26,22 +23,31 @@ internal class ServiceProviderBuilder
 
         public ServiceProviderBuilder(string logFilePath = null, IConsole console = null)
         {
-            LogManager.Configuration = BuildNLogConfiguration(logFilePath);
-            LogManager.ReconfigExistingLoggers();
+            var disableTelemetryEnvVariableValue = Environment.GetEnvironmentVariable(
+               LoggingConstants.OryxDisableTelemetryEnvironmentVariableName);
+            _ = bool.TryParse(disableTelemetryEnvVariableValue, out bool disableTelemetry);
+            var config = new TelemetryConfiguration();
+            var aiKey = disableTelemetry ? string.Empty : Environment.GetEnvironmentVariable(
```

Comment:
Would it be possible to use `null` if the telemetry is disabled, rather than `string.Empty`? It feels like `null` would better represent what we mean, but maybe there are some challenges coming with it

Reply:
Actually do we even need this extra variable? could we just replace this with:
```
if (!disableTelemetry)
{
    config.ConnectionString = Environment.GetEnvironmentVariable(LoggingConstants.ApplicationInsightsConnectionStringKeyEnvironmentVariableName);
}
```

Reply:
With the new changes I checked in, looks like we need an additional variable for aiKey to check if we need to send telemetry to AI.
The method `AddApplicationInsights` throws an error if Connection String is empty or null.

Reply:
> Would it be possible to use `null` if the telemetry is disabled, rather than `string.Empty`? It feels like `null` would better represent what we mean, but maybe there are some challenges coming with it

Makes sense! I do not see any issue if we use null instead of string.Empty here.